documentation/source/best_practices.rst

```diff
@@ -0,0 +1,50 @@
+.. _best_practices:
+
+**************
+Best Practices
+**************
+
+.. _performance:
+
+Performance Considerations
+==========================
+
+In order to get the maximum performance out of a cocotb testbench,
+the following points may help you.
+
+
+Clock Generator in HDL
+----------------------
+
+Generate your clock with a HDL module instead of using :class:`~cocotb.clock.Clock`.
+Doing so prevents calls back into Python every single clock half-period.
```

Comment:
It might be worth expanding this. For example, what are best practices about how to implement that? I'd recommend they use a wrapper instead of adding behavior clock generation to their design. And that has implications, `TOPLEVEL` will have to change, and so the dut parameter will change as well.

Maybe that's all obvious?

Reply:
Not obvious, we should actually have a whole example for this.
I'll see what I can do.